framework.go

```diff
@@ -0,0 +1,171 @@
+package clustertest
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/giantswarm/clustertest/pkg/application"
+	"github.com/giantswarm/clustertest/pkg/client"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	applicationv1alpha1 "github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	cr "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	KubeconfigEnvVar = "E2E_KUBECONFIG"
+)
+
+// Framework is the overall framework for testing of clusters
+type Framework struct {
+	DisableLogging bool
+
+	kubeconfigPath string
+	mcClient       *client.Client
+	wcClients      map[string]*client.Client
```

Comment:
Which use case do you have in mind that requires having several workload cluster clients?

Reply:
Nothing concrete in mind but I seem to recall it mentioned at some point. Do you think this complicates things too much?

Reply:
I would prefer to leave this out until we have a specific use case / test scenario that we want to write. I don't see why testing ingress or networking would require more than 1 workload cluster.